packages/utils/src/getTemplate.ts

```diff
@@ -18,6 +18,17 @@ export default function getTemplate<
   if (name === 'ButtonTemplates') {
     return templates[name];
   }
+  // Allow templates to be customized per-field by using string keys from the registry
+  if (
+    Object.hasOwn(uiOptions, name) &&
+    typeof uiOptions[name] === 'string' &&
+    Object.hasOwn(templates, uiOptions[name] as string)
+  ) {
+    const key = uiOptions[name];
```

Comment:
To truly support things in a way that matches how `widgets` and `fields` updating the `TemplateTypes` type in `@rjsf/utils/src/types.ts` to add this one additional line is all the extra work you will need to do:

```
  /** Allow this to support any named `ComponentType` or an object of named `ComponentType`s */
  [name: string]: ComponentType<any> | { [name: string]: ComponentType<any> };
```

Reply:
Oops, I was wrong. It will require creating a new interface called `RJSFBaseProps` with the definition:
```
export interface RJSFBaseProps<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any> {
  /** The schema object for the field being described */
  schema: S;
  /** The uiSchema object for this description field */
  uiSchema?: UiSchema<T, S, F>;
  /** The `registry` object */
  registry: Registry<T, S, F>;
}
```

Then you will want to update all of the interfaces and types that the `TemplateTypes` inner props that have `ComponentType<XXX>` to make the `XXX` interface/type extend `RJSFBaseProps`

Then add the following to `TemplateTypes` instead of what I mentioned above:

/** Allow this to support any named `ComponentType` or an object of named `ComponentType`s */
  [name: string]: ComponentType<RJSFBaseProps<T, S, F> | { [name: string]: ComponentType<RJSFBaseProps<T, S, F> };